Potentially interesting modifications to FibbingNode

lferran opened the pull request.

Commit:
resynced Quagga submodule with own fork!

Commit:
allowed LSA throttling configuration

Commit:
allowed LSA throttling configuration

Commit:
adding a few useful functions to the API

Commit:
adding a few useful functions to the API

Commit:
Flushing routes from old sessions when connecting again to controller

Commit:
Added UDP black hole for UDP flows

Commit:
Added counter collector call and removal

Commit:
submodule updated

Commit:
few changes

Commit:
introduced previous modifications

Commit:
logs LS Sequence Number - but still has to be improved

Commit:
small changes

Commit:
integrating new features previously developed in new branch

Commit:
Integrated quagga patch for faster OSPF convergence into Fibbing code

Commit:
cherry picking useful commits from the previous ones

Commit:
starting to deal with LS seq_num in lsdb.py

Commit:
seq_num should be removed too when LS is removed from the database

Commit:
seq_num should be removed too when LS is removed from the database

Commit:
updated submodule

oliviertilmans commented on .gitmodules:
```diff
@@ -1,3 +1,3 @@
 [submodule "Quagga"]
 	path = Quagga
-	url = https://github.com/Fibbing/Quagga.git
+	url = https://github.com/lferran/Quagga.git
```
This is something we cannot merge

lferran replied:
True. I wanted to select specific commits for the pull request, and obviously leave this one out.. but I didn't find how.

oliviertilmans commented on fibbingnode/misc/mininetlib/ipnet.py:
```diff
@@ -48,7 +48,7 @@ def __init__(self,
                  private_ip_net='10.0.0.0/8',
                  controller_net='172.16.0.0/12',
-                 ipBase='192.168.0.0/16',
+                 ipBase='192.0.0.0/8',
```
Changing that default looks hazardous as 192/8 is part of the public space

lferran replied:
:+1: 
When I was creating a fat-tree topology with k > 4, the IPNet constructor was running out of broadcast domains... and this was just a quick fix when attempting to solve the problem.

What would be the best way to have more addresses? increasing the max prefix len?

oliviertilmans commented on fibbingnode/misc/mininetlib/ipnet.py:
```diff
@@ -155,6 +155,17 @@ def start(self, timeout=None):
+        log.info('*** Adding IPTABLES RULES\n')
+        for h in self.hosts:
+            # Drop all UDP traffic
+            cmd = "iptables -A INPUT -s {0} -p udp -j DROP".format(self.ipBase)
```
What is the use case for dropping UDP traffic ?

lferran replied:
This is just a black hole (ie consumer) at the Mininet hosts, useful for the creation of UDP flows in the network.

Sorry, this shouldn't be here, but in the TE code instead. :+1:

oliviertilmans commented on fibbingnode/misc/mininetlib/ipnet.py:
```diff
@@ -423,9 +452,10 @@ def _add_node(self, n, props):
             if not nh:
                 continue  # Skip loopback and the likes
-            props[nh.node.name] = {
+            props[itf.name] = {
```
Why is that change needed ? It prevents from accessing link information easily, e.g. db[a][b][ip] to know the ip address of a towards b.

oliviertilmans commented on fibbingnode/misc/router.py:
```diff
@@ -21,6 +21,7 @@
 OSPFD_PORT = 2604
 OSPFD_PASSWORD = 'zebra'
 
+from fibte import counterCollector_path, ifDescrNamespace_path, tmp_files
```
This set of changes should not be here as they depend on some external lib

oliviertilmans commented on fibbingnode/res/default.cfg:
```diff
@@ -5,12 +5,25 @@ cost=1
 dead_interval=40
 # ospf hello interval timer
-hello_interval=10
+hello_interval=20
```
This will slow down _a lot_ the convergence time at bootstrap, which is not something we want for quick experiments

oliviertilmans commented on fibbingnode/southbound/fibbing.py:
```diff
@@ -323,6 +323,14 @@ def proxy_remove(self, points):
     def proxy_connected(self, session):
         self.root.send_lsdblog_to(session)
 
+        if self.routes:
```
This change would be super useful!
Could you try to:
- Limit the route removal to only the ones that were advertized by the application.
- Perform that removal when an application disconnects

oliviertilmans commented on fibbingnode/southbound/lsdb.py:
```diff
@@ -430,23 +440,53 @@ def forwarding_address_of(self, src, dst):
     def process_lsa(self):
         while self.keep_running:
-            commit = False
+            cosmmit = False
```
This looks like a typo to me

lferran replied:
yep ;)

oliviertilmans commented:
Could you try to address the comments and then rebase to get a cleaner history?
Thanks!

Commit:
branch set to initial FibbingNode state

Commit:
extended a bit the southbound api

Commit:
parsing LSA seq_num into lsdb

lferran commented:
ok! that's what I did:
- Created a new branch new-fibbing-features
- Checked out to the new branch
- Revert all the commits done so far (so it looks exactly like upstream, in this case: Fibbing/FibbingNode)
- Then I cherry-picked some of the useful commits from the previous pull request!

I will finish addressing all the comments you told me, and then we could just create another pull-request from the new branch.

From now on, I will use this branch to change stuff that could be interesting to you. And keep weird fixes in my master for now...

What do you think? Please let me know if you think I could organize it in a better way.

oliviertilmans commented:
This is indeed a much cleaner way to handle this !

Commit:
merged conflicts when merging dev branch into master

Commit:
small changes

Commit:
Merge branch 'new-fibbing-features'

Commit:
addressed some of the pull request issues

Commit:
addressed some of the pull request issues

Commit:
removed fibte dependency

Commit:
removed newlines

oliviertilmans commented:
Closing as per #5
